rollup/types.go

```diff
@@ -404,13 +420,73 @@ func (c *CTCTxEIP155) Decode(b []byte) error {
 		return errors.New("CTCTxEIP155 decoding overflow")
 	}
 	copy(c.Signature[:], b[:65])
-	c.gasLimit = binary.BigEndian.Uint16(b[65 : 65+2])
-	c.gasPrice = b[67]
-	hi := uint32(b[68])
-	lo := uint32(binary.BigEndian.Uint16(b[69 : 69+2]))
+	hi := uint32(b[65])
+	lo := uint32(binary.BigEndian.Uint16(b[66:68]))
+	c.gasLimit = (hi << 16) | lo
+	hi = uint32(b[68])
+	lo = uint32(binary.BigEndian.Uint16(b[69:71]))
+	c.gasPrice = (hi << 16) | lo
+	hi = uint32(b[71])
+	lo = uint32(binary.BigEndian.Uint16(b[72:74]))
+	c.nonce = (hi << 16) | lo
+	c.target = common.BytesToAddress(b[74:94])
+	c.data = b[94:]
+	return nil
+}
+
+type CTCTxEthSign struct {
+	Signature [65]byte
+	gasLimit  uint32 // uint24
+	gasPrice  uint32 // uint24
+	nonce     uint32 // uint24
+	target    common.Address
+	data      []byte
+}
```

Comment:
Wondering if we can instead just have a single `CTCTxEIP155` struct that also has an `isEthSign boolean` field. But just a nice to have, seems fine for now!

Reply:
That would help to remove some duplication for sure. This could be done by using the `CTCTransaction` `typ` property and decoupling the `tx` struct from the `typ` field, so that multiple `typ`s can share the same `tx` implementation.